src/Analyzers/CSharp/CodeFixes/UseIndexOrRangeOperator/CSharpUseRangeOperatorCodeFixProvider.cs

```diff
@@ -99,23 +99,28 @@ private static InvocationExpressionSyntax GetInvocationExpression(Diagnostic d,
         private static ExpressionSyntax FixOne(Result result, SyntaxGenerator generator)
         {
             var invocation = result.Invocation;
-            var expression = invocation.Expression is MemberAccessExpressionSyntax memberAccess
-                ? memberAccess.Expression
-                : invocation.Expression;
 
             var rangeExpression = CreateRangeExpression(result, generator);
             var argument = Argument(rangeExpression).WithAdditionalAnnotations(Formatter.Annotation);
             var arguments = SingletonSeparatedList(argument);
 
             if (result.MemberInfo.OverloadedMethodOpt == null)
             {
+                var expression = invocation.Expression switch
+                {
+                    MemberAccessExpressionSyntax memberAccess => memberAccess.Expression,
+                    MemberBindingExpressionSyntax _ => null,
+                    _ => invocation.Expression
+                };
+
                 var argList = invocation.ArgumentList;
-                return ElementAccessExpression(
-                    expression,
-                    BracketedArgumentList(
+                var argumentList = BracketedArgumentList(
                         Token(SyntaxKind.OpenBracketToken).WithTriviaFrom(argList.OpenParenToken),
                         arguments,
-                        Token(SyntaxKind.CloseBracketToken).WithTriviaFrom(argList.CloseParenToken)));
+                        Token(SyntaxKind.CloseBracketToken).WithTriviaFrom(argList.CloseParenToken));
```

Comment:
i think it would be cleaner/nicer to just unilateraly make the argument list first, then specifically do a check if the expr is a member binding.  If so, you make an ElementBinding.  Then there's no weird tracking of this null value that has special meaning later on.

Reply:
Done in 26cc13fb.